### PR TITLE
Niklaz/sc 23615/sharing groups by email

### PR DESCRIPTION
**Pull request by philosopher-engineer**

### CODE REVIEW

File: openapi-v1.yaml [hunk 1]
```diff
@@ -2677,6 +2679,36 @@ definitions:
...
+          - write
+      group_actions:
+        description: List of permitted group actions
+        type: array
+        items:
+          $ref: "#/definitions/GroupActions"
+        example:
+          - read
+          - write
+      invitee_email_or_name:
```
Comment (antalakas) [inline openapi-v1.yaml:2706]: invitee_email_or_namespace

File: openapi-v1.yaml [hunk 2]
```diff
@@ -6718,6 +6768,44 @@ paths:
           schema:
             $ref: "#/definitions/Error"
 
+  /invitations/{namespace}/{group}/share/group:
```
Comment (antalakas) [inline openapi-v1.yaml:6771]: This should be `/invitations/{namespace}/{group}/share`

File: openapi-v1.yaml [hunk 3]
```diff
@@ -6718,6 +6768,44 @@ paths:
...
+        description: name/uri of group that is url-encoded
+        type: string
+        required: true
+    post:
+      tags:
+        - invitation
+      description: Sends email to multiple recipients with sharing information regarding a group
+      operationId: shareGroupByInvite
+      parameters:
+        - name: group_email_invite
```
Comment (antalakas) [inline openapi-v1.yaml:6789]: Let's make it `email_invite` same as in other invitation related handlers

File: openapi-v1.yaml [hunk 4]
```diff
@@ -6718,6 +6768,44 @@ paths:
...
+          in: body
+          description: list of email/name recipients
```
Comment (antalakas) [inline openapi-v1.yaml:6791]: namespace

File: openapi-v1.yaml [hunk 5]
```diff
@@ -6718,6 +6768,44 @@ paths:
...
+          schema:
+            $ref: "#/definitions/InvitationGroupShareEmail"
+          required: true
+      responses:
+        204:
+          description: Email sent successfully to user for email confirmation link
```
Comment (antalakas) [inline openapi-v1.yaml:6797]: Maybe rephrase to

Email sent successfully to user with an email confirmation link

**Review by antalakas — CHANGES_REQUESTED** [review]

File: openapi-v1.yaml (hunk 2 above)
Comment (philosopher-engineer) [reply openapi-v1.yaml:6771]: If omit the /share then we have same path with /invitations/{namespace}/{array}/share

Comment (antalakas) [reply openapi-v1.yaml:6771]: Let's reuse `/invitations/{namespace}/{array}/share`, updating param list with InvitationGroupShareEmail, REST should be able to distinguish if it is an array or a group and act accordingly

**Review by antalakas — CHANGES_REQUESTED** [review]

File: openapi-v1.yaml [hunk 6]
```diff
@@ -2677,6 +2679,36 @@ definitions:
         items:
           type: string
 
+  InvitationGroupShareEmail:
+    description: Encapsulates information regarding inviting people using email to share group, same permissions for all invitees
+    type: object
+    required:
+      - array_actions
+      - group_actions
+      - invitee_email_or_name
```
Comment (alekshs) [inline openapi-v1.yaml:2688]: Typo? the actual prop is `invitee_email_or_namespace ` not `invitee_email_or_name`.

**Review by alekshs — CHANGES_REQUESTED** [review]

File: openapi-v1.yaml (hunk 6 above)
Comment (philosopher-engineer) [reply openapi-v1.yaml:2688]: Namespace is an object according to my understanding "a space (an object) characterized by a name". This was understood by code: `// NamespaceName NamespaceName
// swagger:model NamespaceName
type Namespace struct {

	// unique id of namepace, user or organization
	UUID *uuid.UUID `json:"id,omitempty" gorm:"PRIMARY_KEY;size:36"`

	// user or organization name must be unique
	// Required: true
	// Max Length: 20
	// Min Length: 4
	// Pattern: ^[\w.\-]+$
	NamespaceName string `json:"namespace_name" gorm:"unique_index"`

	User     User `json:"-" gorm:"association_autocreate:false;association_autoupdate:false;foreignkey:UserUUID"`
	UserUUID *uuid.UUID

	Organization     Organization `json:"-" gorm:"association_autocreate:false;association_autoupdate:false;foreignkey:OrganizationUUID"`
	OrganizationUUID *uuid.UUID

	CreatedAt time.Time  `json:"-"`
	UpdatedAt time.Time  `json:"-"`
	DeletedAt *time.Time `json:"-"`
}`

So I though that a user or an organization is a namespace. 
The invitee_email_or_name field should there be like invitee_email_or_namespacename (namespace exists in namespace struct above) so I chose name as a shorter word for namespace name. What do you think @alekshs ?

Comment (alekshs) [reply openapi-v1.yaml:2688]: Both names make sense. We just need to make things consistent. They aren't consistent at the moment and as a result client generators throw an error.

Comment (philosopher-engineer) [reply openapi-v1.yaml:2688]: ok changing this to namespace

Comment (philosopher-engineer) [reply openapi-v1.yaml:2688]: Sorry @alekshs did not see that invitee_email_or_namespace was used below. I am fixing this.

**Review by antalakas — APPROVED** [review]

**Review by antalakas — APPROVED** [review]
